components/server/ee/src/prebuilds/gitlab-app.ts

```diff
@@ -33,28 +33,36 @@ export class GitLabApp {
     @postConstruct()
     protected init() {
         this._router.post("/", async (req, res) => {
-            const event = req.header("X-Gitlab-Event");
-            if (event === "Push Hook") {
-                const context = req.body as GitLabPushHook;
-                const span = TraceContext.startSpan("GitLapApp.handleEvent", {});
-                span.setTag("request", context);
-                log.debug("GitLab push hook received", { event, context });
-                let user: User | undefined;
-                try {
-                    user = await this.findUser({ span }, context, req);
-                } catch (error) {
-                    log.error("Cannot find user.", error, {});
-                }
-                if (!user) {
-                    res.statusCode = 503;
-                    res.send();
-                    return;
+            try {
+                const event = req.header("X-Gitlab-Event");
```

Comment:
Does Gitlab send any kind of signature with the payload that we should verify, like [Github does](https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks#validating-payloads-from-github)?

Reply:
Nope, see https://gitlab.com/gitlab-org/gitlab/-/issues/19367